### ISSUE TITLE
shell script which is also valid yaml is rewrapping when updating with a prefix

### ISSUE THREAD

**Issue by joemaller** [joemaller/version-everything#40]
Found this in the bump-php.sh script in [ideasonpurpose/docker-phpunit-watch](https://github.com/ideasonpurpose/docker-phpunit-watch), text files (maybe more?) are being pre-processed as something other than text which is causing them to re-wrap. I think they're being treated as yaml since the **bump-phpunit.sh** test file parses as valid yaml.

**Comment by joemaller** [comment]
This seems to be happening because the file is valid YAML. The first step of the non-matched-extension checks is to try the file as json, xml and yaml, _assuming_ each will fail as invalid. Because the test shell script validates, it gets handled as YAML, but because there's a prefix, [`bumpYAML()`](https://github.com/joemaller/version-everything/blob/979b41726ec9e3a1c4ee6a4cd67ce9e5e6659f87/app/lib/bump-yaml.js) first correctly [processes the file as plaintext](https://github.com/joemaller/version-everything/blob/979b41726ec9e3a1c4ee6a4cd67ce9e5e6659f87/app/lib/bump-yaml.js#L19-L22):

```js
  if (config.prefixes.length > 0) {
    prefixData = bumpPlainText(data, version, config);
    hasVersion = !!prefixData?.data;
  }
```

then run the file through the [YAML parser](https://github.com/joemaller/version-everything/blob/979b41726ec9e3a1c4ee6a4cd67ce9e5e6659f87/app/lib/bump-yaml.js#L25) anyway, which, because it's valid, doesn't throw an error:

```js
  const yamlObj = YAML.parse(prefixData.data || data);
```


and finally [returns the parsed, reformatted YAML](https://github.com/joemaller/version-everything/blob/979b41726ec9e3a1c4ee6a4cd67ce9e5e6659f87/app/lib/bump-yaml.js#L39-L42):
```js
  return {
    oldVersion: oldVersion,
    data: yamlData.toString(config.yaml),
  };```

**Comment by joemaller** [comment]
Closed along with #41 6434a01858befb173e57b24ad29fbd982e15aee9
